presto-orc/src/main/java/com/facebook/presto/orc/DictionaryCompressionOptimizer.java

```diff
@@ -73,9 +109,14 @@ public DictionaryCompressionOptimizer(
         this.stripeMaxRowCount = stripeMaxRowCount;
 
         checkArgument(dictionaryMemoryMaxBytes >= 0, "dictionaryMemoryMaxBytes is negative");
+        checkArgument(dictionaryMemoryAlmostFullRangeBytes >= 0, "dictionaryMemoryRangeBytes is negative");
         this.dictionaryMemoryMaxBytesHigh = dictionaryMemoryMaxBytes;
-        this.dictionaryMemoryMaxBytesLow = (int) Math.max(dictionaryMemoryMaxBytes - DICTIONARY_MEMORY_MAX_RANGE.toBytes(), 0);
+        this.dictionaryMemoryMaxBytesLow = Math.max(dictionaryMemoryMaxBytes - dictionaryMemoryAlmostFullRangeBytes, 0);
```

Comment:
I see that we use `Max(... - ..., 0)`, what do you think about enforcing this delta to be >=0 instead?

Reply:
I did that and there were test failures :(
There are tests, which define the dictionary memory to 1MB and for backward compat, we want it to be 4MB.